tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/multiply-operation.bal

```diff
@@ -1189,6 +1189,20 @@ function testNoShortCircuitingInMultiplicationWithNonNullable() {
     assertEqual(intVal, 30);
 }
 
+function testDecimalMultiplicationUnderflow() {
+    decimal d1 = 9.999999999999999999999999999999999E-6001;
+    decimal d2 = 1E-143;
+    assertEqual(d1 * d2, 0.0d);
+
+    d1 = -9.999999999999999999999999999999999E-6001;
+    d2 = 1E-143;
+    assertEqual(d1 * d2, 0.0d);
+
+    d1 = -9.999999999999999999999999999999999E-6101;
+    d2 = 1E-76;
+    assertEqual(d1 * d2, 0.0d);
```

Comment:
What should be the precision of the resulting decimal value here? Shall we use `assertEqual` here and use assertTrue with `===` operator here to check the decimal value (Can also add statement with `==` if needed) https://ballerina.io/spec/lang/master/#section_6.29

Reply:
The above operations are result in values which are not belonged to the normal decimal values set (they are decimal subnormals and underflow according to the IEEE-754 Decimal128 spec) where the precision is lost. According to https://github.com/ballerina-platform/ballerina-spec/issues/995 subnormal values also need to be treated as decimal value Zero.

Reply:
Agreed to use decimal `0d` as the subnormal decimal zero. Hence marking this conversation as resolved.